extensions/functions_comparison.yaml

```diff
@@ -215,3 +215,46 @@ scalar_functions:
         variadic:
           min: 2
         return: any1
+  -
+    name: "least"
+    description: >-
+      Returns the smallest value.  The function will return null if any argument evaluates
+      to null.
+
+      String comparison is done in lexicographical ordering, one character at a time, from
+      left to right. Uppercase letters are less than lowercase letters.
+    impls:
+      - args:
+          - value: T
+        variadic:
+          min: 2
+        return: T|?
+  -
+    name: "least_skip_null"
+    description: >-
+      Returns the smallest value. The function will return null only if all arguments evaluate
+      to null.
+
+      String comparison is done in lexicographical ordering, one character at a time, from
+      left to right. Uppercase letters are less than lowercase letters.
+    impls:
+      - args:
+          - value: T
+        variadic:
+          min: 2
+        return: T&?
+  -
+    name: "greatest"
+    description: >-
+      Returns the largest value. Only return null if 'all' arguments evaluate to null.
+
+      String comparison is done in lexicographical ordering, one character at a time, from left to right.
+      Uppercase letters are less than lowercase letters.
+
+      There is no greatest_skip_null function because it behaves the same as greatest.
```

Comment:
I disagree with this conclusion.  There is no existing engine I am aware of that behaves this way.  For example, in both Oracle and MySQL `GREATEST(1, 3, NULL)` yields `NULL`.  The theory is that "if any one of the inputs is unknown then I cannot know which is the greatest value because it may be the unknown one"

I think we do need a `greatest_skip_null` variant and this variant should not skip nulls.